src/components/Task.tsx

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import '../css/index.css'
+import { Tasks, useReminders } from '../utils/RemindersContext'
+import calendar from '../assets/icons/calendar-two.png'
+import check from '../assets/icons/checkmark.png'
+import flag from '../assets/icons/image.png'
+
+const Task = ({ id, input, flagged }: Tasks) => {
+  const {
+    activeList,
+    setLists,
+    lists,
+    activeListId,
+  } = useReminders();
+
+  if (!activeList) {
+    return <main className="w-[65%] ml-16 mt-8" />;
+  }
+
+  return (
+    <div className="md:ml-6 mt-6 md:w-[46%] flex flex-row">
+      <button
+      type="button"
+      onClick={() => setLists(lists.map(list =>
+        list.id === activeListId
+          ? {
+            ...activeList,
+            tasks: activeList.tasks.filter(
+              task => task.id !== id
+            ),
+          }
+          : list)
+      )}
```

Comment:
When event handlers are as complex as this (with specific meaning) it's more maintanble/readable to move them into their own function and pass them in as a prop.

I.e in this case:
```
const onClickRemove = useCallback(...)
...
onClick={onClickRemove}
```